redshift_connector/cursor.py

```diff
@@ -239,6 +239,51 @@ def executemany(self: "Cursor", operation, param_sets) -> "Cursor":
         self._row_count = -1 if -1 in rowcounts else sum(rowcounts)
         return self
 
+    def insert_data_bulk(self: "Cursor", filename,table_name, column_indexes,column_names,delimeter)-> "Cursor":
+  
+        """runs a single bulk insert statement into the database.
+           
+           This method is part of the `DBAPI 2.0 specification
+           <http://www.python.org/dev/peps/pep-0249/>`_.
+
+            :param filename: str
+                The name of the file to read from.
+            :param table_name: str
+                The name of the table to insert to.
+            :param column_names:list
+                The name of the columns in the table to insert to.  
+            :param column_indexes:list
+                The indexes of the columns in the table to insert to.
+            :param delimeter: str
+                The delimeter to use when reading the file.
+
+            Returns
+            
+            -------
+            The Cursor object used for executing the specified database operation: :class:`Cursor`
+
+        """
+      
+        sql_query = f"INSERT INTO  {table_name} ("
+        for column_name in column_names:
+            sql_query = sql_query + f", {column_name}"
+        
+        sql_query = sql_query + ") VALUES %s"
+
+        with open(filename) as csvfile:
+            reader = csv.reader(csvfile)
+            next(reader)
+            values_list = []
+            
+            for row in reader:
+                values=row.split(delimeter)
+                param_set=[]
+                for index in column_indexes:
+                    param_set.append(values[index])
+                values_list.append(tuple(param_set))
+            self.execute(sql_query, values_list)
+        return self
+
```

Comment:
This method would use `information_schema.columns` to validate that the columns provided exist in the table the user specified.
```suggestion


    def __has_valid_columns(self: "Cursor", table: str, columns: typing.List[str]) -> bool:
        split_table_name: typing.List[str] = table.split(".")
        q: str = "select 1 from information_schema.columns where table_name = ? and column_name = ?"

        if len(split_table_name) == 2:
            q += " and table_schema = ?"
            param_list = [[split_table_name[1], c, split_table_name[0]] for c in columns]
        else:
            param_list = [[split_table_name[0], c] for c in columns]

        temp = self.paramstyle
        self.paramstyle = "qmark"

        try:
            for params in param_list:
                self.execute(q, params)
                res = self.fetchone()
                if res[0] != 1:
                    raise InterfaceError("Invalid column name: {} specified for table: {}".format(params[1], table))
        except:
            raise
        finally:
            # reset paramstyle to it's original value
            self.paramstyle = temp

        return True
```

Reply:
addressed in 43e0cf6